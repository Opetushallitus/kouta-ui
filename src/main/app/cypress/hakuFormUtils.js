import merge from 'lodash/merge';

import organisaatio from './data/organisaatio';
import {
  stubKoodistoRoute,
  stubHakemuspalveluLomakkeetRoute,
  stubOppijanumerorekisteriHenkiloRoute,
} from './utils';

export const stubHakuFormRoutes = ({ cy, organisaatioOid }) => {
  cy.server();

  cy.route({
    method: 'GET',
    url: `**/organisaatio-service/rest/organisaatio/v4/${organisaatioOid}**`,
    response: merge(organisaatio(), {
      oid: organisaatioOid,
    }),
  });

  cy.route({
    method: 'POST',
    url: '**/organisaatio-service/rest/organisaatio/v4/findbyoids',
    response: [
      merge(organisaatio(), {
        oid: organisaatioOid,
      }),
    ],
  });

  stubKoodistoRoute({ koodisto: 'haunkohdejoukko', cy });
  stubKoodistoRoute({ koodisto: 'hakutapa', cy });
  stubKoodistoRoute({ koodisto: 'kausi', cy });
  stubKoodistoRoute({ koodisto: 'opetuspaikkakk', cy });
  stubKoodistoRoute({ koodisto: 'kausi', cy });

  cy.route({
    method: 'GET',
    url: '**/haku/list**',
    response: [],
  });

  stubHakemuspalveluLomakkeetRoute({ cy });
  stubOppijanumerorekisteriHenkiloRoute({ cy });
};
