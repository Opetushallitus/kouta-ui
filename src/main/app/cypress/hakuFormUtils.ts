import createKoodisto from '#/cypress/data/koodisto';

import organisaatio from '#/cypress/data/organisaatio';
import createKoodi from '#/cypress/data/koodi';

import {
  stubKoodistoRoute,
  stubHakemuspalveluLomakkeetRoute,
  stubOppijanumerorekisteriHenkiloRoute,
  stubKoodiRoute,
  stubCommonRoutes,
} from './utils';

export const stubHakuFormRoutes = ({ organisaatioOid }) => {
  stubCommonRoutes();

  cy.route({
    method: 'GET',
    url: `**/organisaatio-service/rest/organisaatio/v4/${organisaatioOid}**`,
    response: organisaatio({
      oid: organisaatioOid,
    }),
  });

  cy.route({
    method: 'POST',
    url: '**/organisaatio-service/rest/organisaatio/v4/findbyoids',
    response: [
      organisaatio({
        oid: organisaatioOid,
      }),
    ],
  });

  stubKoodistoRoute({ koodisto: 'haunkohdejoukontarkenne' });
  stubKoodistoRoute({ koodisto: 'kausi' });
  stubKoodistoRoute({ koodisto: 'opetuspaikkakk' });
  stubKoodistoRoute({ koodisto: 'kausi' });
  stubKoodistoRoute({ koodisto: 'posti' });
  stubKoodiRoute(createKoodi({ koodisto: 'posti', versio: 2 }));

  cy.route({
    method: 'GET',
    url: `**/koodisto-service/rest/json/haunkohdejoukko/koodi**`,
    response: createKoodisto({ koodisto: 'haunkohdejoukko' }).map((v, index) =>
      index === 0
        ? {
            ...v,
            koodiUri: 'haunkohdejoukko_12',
          }
        : v
    ),
  });

  cy.route({
    method: 'GET',
    url: `**/koodisto-service/rest/json/hakutapa/koodi**`,
    response: createKoodisto({ koodisto: 'hakutapa' }).map((v, index) =>
      index === 0
        ? {
            ...v,
            koodiUri: 'hakutapa_01',
          }
        : v
    ),
  });

  cy.route({
    method: 'GET',
    url: '**/haku/list**',
    response: [],
  });

  cy.route({
    method: 'GET',
    url: `**/toteutus/list**`,
    response: [],
  });

  stubHakemuspalveluLomakkeetRoute();
  stubOppijanumerorekisteriHenkiloRoute();
};
