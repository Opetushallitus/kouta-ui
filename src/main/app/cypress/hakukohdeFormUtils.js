import merge from 'lodash/merge';

import { stubKoodistoRoute } from './utils';
import organisaatio from './data/organisaatio';
import haku from './data/haku';

export const stubHakukohdeFormRoutes = ({ cy, organisaatioOid, hakuOid }) => {
  cy.server();

  cy.route({
    method: 'GET',
    url: `**/organisaatio-service/rest/organisaatio/v4/${organisaatioOid}**`,
    response: merge(organisaatio(), {
      oid: organisaatioOid,
    }),
  });

  stubKoodistoRoute({ koodisto: 'pohjakoulutusvaatimustoinenaste', cy });
  stubKoodistoRoute({ koodisto: 'kausi', cy });
  stubKoodistoRoute({ koodisto: 'valintakokeentyyppi', cy });
  stubKoodistoRoute({ koodisto: 'liitetyypitamm', cy });

  cy.route({
    method: 'GET',
    url: `**/haku/${hakuOid}`,
    response: merge(haku(), {
      oid: hakuOid,
      organisaatioOid: organisaatioOid,
    }),
  });

  cy.route({
    method: 'GET',
    url: `**/hakukohde/list**`,
    response: [],
  });
};
