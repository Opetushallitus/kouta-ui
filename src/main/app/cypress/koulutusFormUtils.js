import merge from 'lodash/merge';

import { stubKoodistoRoute } from './utils';
import organisaatioHierarkia from './data/organisaatioHierarkia';
import organisaatio from './data/organisaatio';
import koodisto from './data/koodisto';

export const stubKoulutusFormRoutes = ({ cy, organisaatioOid }) => {
  cy.server();

  cy.route({
    method: 'GET',
    url: `**/organisaatio-service/rest/organisaatio/v4/hierarkia/hae?oid=${organisaatioOid}**`,
    response: organisaatioHierarkia({ rootOid: organisaatioOid }),
  });

  cy.route({
    method: 'GET',
    url: `**/organisaatio-service/rest/organisaatio/v4/${organisaatioOid}**`,
    response: merge(organisaatio(), {
      oid: organisaatioOid,
    }),
  });

  cy.route({
    method: 'POST',
    url: '**/organisaatio-service/rest/organisaatio/v4/findbyoids',
    response: [
      merge(organisaatio(), {
        oid: organisaatioOid,
      }),
    ],
  });

  cy.route({
    method: 'GET',
    url:
      '**/koodisto-service/rest/json/relaatio/sisaltyy-ylakoodit/koulutustyyppi_*',
    response: koodisto({ koodisto: 'koulutus' }),
  });

  cy.route({
    method: 'GET',
    url: `**/koulutus/list?organisaatioOid=${organisaatioOid}`,
    response: [],
  });

  cy.route({
    method: 'GET',
    url: '**/koodisto-service/rest/codeelement/koulutus_0',
    response: [
      {
        koodiUri: 'koulutus_0',
        versio: 1,
        metadata: [{ kieli: 'fi', nimi: 'Nimi' }],
      },
    ],
  });

  cy.route({
    method: 'GET',
    url:
      '**/eperusteet-service/api/perusteet?tuleva=true&siirtyma=false&voimassaolo=true&poistunut=false&kieli=fi&koulutuskoodi=koulutus_0',
    response: {
      data: [
        {
          kuvaus: { fi: 'koulutus_0 kuvaus' },
          osaamisalat: [],
          tutkintonimikeKoodiUri: 'nimike_1#1',
          id: '1',
        },
      ],
    },
  });

  cy.route({
    method: 'GET',
    url:
      '**/koodisto-service/rest/json/relaatio/sisaltyy-alakoodit/koulutus_0**',
    response: [
      {
        koodiUri: 'okmohjauksenala_1',
        metadata: [{ kieli: 'fi', nimi: 'Ala' }],
      },
      {
        koodiUri: 'opintojenlaajuus_1',
        metadata: [{ kieli: 'fi', nimi: '180' }],
      },
      {
        koodiUri: 'opintojenlaajuusyksikko_1',
        metadata: [{ kieli: 'fi', nimi: 'op' }],
      },
    ],
  });

  stubKoodistoRoute({ koodisto: 'tutkintonimikekk', cy });
  stubKoodistoRoute({ koodisto: 'opintojenlaajuus', cy });
  stubKoodistoRoute({ koodisto: 'koulutuksenjarjestamisenlisaosiot', cy });
};
