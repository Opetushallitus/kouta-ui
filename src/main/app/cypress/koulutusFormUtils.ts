import { merge } from 'lodash';

import {
  stubKoodistoRoute,
  stubOppijanumerorekisteriHenkiloRoute,
  stubEPerusteetByKoulutuskoodiRoute,
  stubCommonRoutes,
} from '#/cypress/utils';
import organisaatioHierarkia from '#/cypress/data/organisaatioHierarkia';
import organisaatio from '#/cypress/data/organisaatio';
import koodisto from '#/cypress/data/koodisto';
import koodistoKoulutus351107 from './data/koodistoKoulutus351107';
import koodistoKoulutukset from './data/koodistoKoulutukset';
import koodistoKoulutus351107Alakoodit from './data/koodistoKoulutus351107Alakoodit';
import koodistoKoulutus309902 from './data/koodistoKoulutus309902';
import koodistoKoulutus309902Alakoodit from './data/koodistoKoulutus309902Alakoodit';
import koodistoKoulutus671112 from './data/koodistoKoulutus671112';
import koodistoKoulutus671112Alakoodit from './data/koodistoKoulutus671112Alakoodit';
import ePerustePerusteenosat6777674 from './data/ePerustePerusteenosat6777674';
import ePerustePerusteenosat6778360 from './data/ePerustePerusteenosat6778360';
import ePeruste6777660TutkinnonOsat from './data/ePeruste6777660TutkinnonOsat';
import ePeruste6777660Rakenne from './data/ePeruste6777660Rakenne';
import ePeruste6777660 from './data/ePeruste6777660';

export const stubKoulutusFormRoutes = ({ organisaatioOid }) => {
  stubCommonRoutes();

  cy.route({
    method: 'GET',
    url: `**/organisaatio-service/rest/organisaatio/v4/hierarkia/hae**`,
    response: organisaatioHierarkia({ rootOid: organisaatioOid }),
  });

  cy.route({
    method: 'GET',
    url: `**/organisaatio-service/rest/organisaatio/v4/${organisaatioOid}**`,
    response: merge(organisaatio(), {
      oid: organisaatioOid,
    }),
  });

  cy.route({
    method: 'POST',
    url: '**/organisaatio-service/rest/organisaatio/v4/findbyoids',
    response: [
      merge(organisaatio(), {
        oid: organisaatioOid,
      }),
    ],
  });

  cy.route({
    method: 'GET',
    url:
      '**/koodisto-service/rest/json/relaatio/sisaltyy-ylakoodit/koulutustyyppi_*',
    response: koodisto({ koodisto: 'koulutus' }),
  });

  cy.route({
    method: 'GET',
    url: `**/koulutus/list?organisaatioOid=${organisaatioOid}`,
    response: [],
  });

  cy.route({
    method: 'GET',
    url: `**/koulutus/*/toteutukset/list**`,
    response: [],
  });

  cy.route({
    method: 'GET',
    url: '**/koodisto-service/rest/codeelement/koulutus_351107/**',
    response: koodistoKoulutus351107,
  });

  cy.route({
    method: 'GET',
    url:
      '**/koodisto-service/rest/json/relaatio/sisaltyy-alakoodit/koulutus_351107**',
    response: koodistoKoulutus351107Alakoodit,
  });

  cy.route({
    method: 'GET',
    url: '**/koodisto-service/rest/codeelement/koulutus_309902/**',
    response: koodistoKoulutus309902,
  });

  cy.route({
    method: 'GET',
    url:
      '**/koodisto-service/rest/json/relaatio/sisaltyy-alakoodit/koulutus_309902**',
    response: koodistoKoulutus309902Alakoodit,
  });

  cy.route({
    method: 'GET',
    url: '**/koodisto-service/rest/codeelement/koulutus_671112/**',
    response: koodistoKoulutus671112,
  });

  cy.route({
    method: 'GET',
    url:
      '**/koodisto-service/rest/json/relaatio/sisaltyy-alakoodit/koulutus_671112**',
    response: koodistoKoulutus671112Alakoodit,
  });

  cy.route({
    method: 'GET',
    url: '**/koodisto-service/rest/json/koulutus/koodi?onlyValidKoodis=true**',
    response: koodistoKoulutukset,
  });

  cy.route({
    method: 'GET',
    url: '**/eperusteet-service/api/perusteenosat/6777674',
    response: ePerustePerusteenosat6777674,
  });

  cy.route({
    method: 'GET',
    url: '**/eperusteet-service/api/perusteenosat/6778360',
    response: ePerustePerusteenosat6778360,
  });

  cy.route({
    method: 'GET',
    url: '**/eperusteet-service/api/perusteenosat/6778360',
    response: ePerustePerusteenosat6778360,
  });

  cy.route({
    method: 'GET',
    url: '**/tutkinnonosat',
    response: ePeruste6777660TutkinnonOsat,
  });

  cy.route({
    method: 'GET',
    url:
      '**/eperusteet-service/api/perusteet/6777660/suoritustavat/reformi/rakenne',
    response: ePeruste6777660Rakenne,
  });

  cy.route({
    method: 'GET',
    url: '**/eperusteet-service/api/perusteet/6777660',
    response: ePeruste6777660,
  });

  stubEPerusteetByKoulutuskoodiRoute();

  stubKoodistoRoute({ koodisto: 'tutkintonimikekk' });
  stubKoodistoRoute({ koodisto: 'opintojenlaajuus' });
  stubKoodistoRoute({ koodisto: 'koulutuksenlisatiedot' });

  stubKoodistoRoute({
    koodisto: 'kansallinenkoulutusluokitus2016koulutusalataso2',
  });

  stubOppijanumerorekisteriHenkiloRoute();
};
