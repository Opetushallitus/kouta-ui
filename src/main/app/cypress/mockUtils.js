const path = require('path');

const MOCK_PATH = 'cypress/mocks';

export const addMockFileRoutes = fileName => {
  cy.readFile(path.join(MOCK_PATH, fileName)).then(data => {
    data.forEach(({ url, method = 'GET', response }) => {
      // TODO: Mock only on matching response body and use cy.intercept()
      cy.route({
        method,
        url,
        status: response.status,
        response: response.body,
      });
    });
  });
};
