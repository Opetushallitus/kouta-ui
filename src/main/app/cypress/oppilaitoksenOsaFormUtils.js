import merge from 'lodash/merge';

import createKoodi from './data/koodi';
import createOrganisaatio from './data/organisaatio';
import { stubKoodiRoute } from './utils';

export const stubOppilaitosFormRoutes = ({ organisaatioOid }) => {
  stubKoodiRoute(createKoodi({ koodisto: 'posti', versio: 2 }));

  cy.route({
    method: 'POST',
    url: '**/organisaatio-service/rest/organisaatio/v4/findbyoids',
    response: [
      merge(createOrganisaatio(), {
        oid: organisaatioOid,
      }),
    ],
  });
};
