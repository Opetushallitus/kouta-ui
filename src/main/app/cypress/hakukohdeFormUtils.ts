import { merge } from 'lodash';

import {
  stubKoodistoRoute,
  stubHakemuspalveluLomakkeetRoute,
  stubOppijanumerorekisteriHenkiloRoute,
  stubKoodiRoute,
  stubCommonRoutes,
  jatka,
} from '#/cypress/utils';

import haku from '#/cypress/data/haku';
import hakukohde from '#/cypress/data/hakukohde';
import createKoodi from '#/cypress/data/koodi';
import koulutus from '#/cypress/data/koulutus';
import toteutus from '#/cypress/data/toteutus';
import valintaperuste from '#/cypress/data/valintaperuste';
import { addMockFileRoutes } from './mockUtils';

const toimipisteTarjoajat = [
  '1.2.246.562.10.16538823663',
  '1.2.246.562.10.2013081415065445951365',
  '1.2.246.562.10.20485193278',
  '1.2.246.562.10.20866993056',
  '1.2.246.562.10.23017513880',
  '1.2.246.562.10.45854578546',
  '1.2.246.562.10.55355715099',
  '1.2.246.562.10.55711304158',
  '1.2.246.562.10.56139411567',
  '1.2.246.562.10.60465978314',
  '1.2.246.562.10.656909794910',
  '1.2.246.562.10.78321186062',
  '1.2.246.562.10.879177258610',
  '1.2.246.562.10.93115250668',
];

const selectedToimipisteNimi = /stadin ammatti- ja aikuisopisto, myllypuron toimipaikka/i;

const selectedToimipiste = '1.2.246.562.10.45854578546';

export const fillJarjestyspaikkaSection = ({ jatka: jatkaProp = true }) => {
  cy.findByTestId('jarjestyspaikkaOidSection').within(() => {
    toimipisteTarjoajat.forEach(oid => {
      cy.get(`[value="${oid}"]`).should('exist');
      if (oid === selectedToimipiste) {
        cy.get(`[value="${oid}"]`).should('not.be.disabled');
      } else {
        cy.get(`[value="${oid}"]`).should('be.disabled');
      }
    });

    cy.findByText(selectedToimipisteNimi).click();
    if (jatkaProp) {
      jatka();
    }
  });
};

export const prepareTest = ({
  tyyppi,
  hakuOid,
  hakukohdeOid,
  organisaatioOid,
  edit = false,
  tarjoajat,
}) => {
  const toteutusOid = '2.1.1.1.1.1';
  const koulutusOid = '3.1.1.1.1';
  const valintaperusteId = '649adb37-cd4d-4846-91a9-84b58b90f928';

  const testHakukohdeFields = {
    toteutusOid,
    hakuOid,
    organisaatioOid,
    oid: hakukohdeOid,
    valintaperusteId,
  };

  cy.server();

  addMockFileRoutes('hakukohde.mock.json');
  stubHakukohdeFormRoutes({ organisaatioOid, hakuOid });

  cy.route({
    method: 'GET',
    url: `**/toteutus/${toteutusOid}`,
    response: merge(toteutus({ tyyppi }), {
      oid: toteutusOid,
      organisaatioOid,
      koulutusOid: koulutusOid,
      tila: 'julkaistu',
      tarjoajat,
    }),
  });

  cy.route({
    method: 'GET',
    url: `**/koulutus/${koulutusOid}`,
    response: merge(koulutus({ tyyppi }), {
      oid: koulutusOid,
      tarjoajat,
      organisaatioOid: organisaatioOid,
      tila: 'julkaistu',
    }),
  });

  cy.route({
    method: 'GET',
    url: '**/valintaperuste/list**',
    response: [
      merge(valintaperuste(), {
        id: valintaperusteId,
        nimi: { fi: 'Valintaperusteen nimi' },
        tila: 'julkaistu',
      }),
    ],
  });

  cy.route({
    method: 'GET',
    url: `**/valintaperuste/${valintaperusteId}`,
    response: merge(valintaperuste(), {
      id: valintaperusteId,
      nimi: { fi: 'Valintaperusteen nimi' },
      tila: 'julkaistu',
    }),
  });

  cy.route({
    method: 'GET',
    url: `**/hakukohde/${hakukohdeOid}`,
    response: merge(hakukohde(), testHakukohdeFields),
  });

  edit
    ? cy
        .route({
          method: 'POST',
          url: '**/hakukohde',
          response: {
            muokattu: false,
          },
        })
        .as('updateHakukohdeRequest')
    : cy
        .route({
          method: 'PUT',
          url: '**/hakukohde',
          response: {
            oid: hakukohdeOid,
          },
        })
        .as('createHakukohdeRequest');

  cy.visit(
    edit
      ? `/organisaatio/${organisaatioOid}/hakukohde/${hakukohdeOid}/muokkaus`
      : `/organisaatio/${organisaatioOid}/toteutus/${toteutusOid}/haku/${hakuOid}/hakukohde`
  );
};

export const stubHakukohdeFormRoutes = ({ organisaatioOid, hakuOid }) => {
  stubCommonRoutes();

  stubKoodistoRoute({ koodisto: 'pohjakoulutusvaatimustoinenaste' });
  stubKoodistoRoute({ koodisto: 'kausi' });
  stubKoodistoRoute({ koodisto: 'valintakokeentyyppi' });
  stubKoodistoRoute({ koodisto: 'liitetyypitamm' });
  stubKoodistoRoute({ koodisto: 'kausi' });
  stubKoodiRoute(createKoodi({ koodisto: 'posti', versio: 2 }));
  stubKoodiRoute(createKoodi({ koodisto: 'posti', versio: 2 }));
  stubKoodiRoute(
    createKoodi({ koodisto: 'posti', koodiArvo: '00350', versio: 1 })
  );

  cy.route({
    method: 'GET',
    url: `**/haku/${hakuOid}`,
    response: merge(haku(), {
      oid: hakuOid,
      organisaatioOid: organisaatioOid,
      hakulomaketyyppi: 'muu',
      tila: 'julkaistu',
    }),
  });

  cy.route({
    method: 'GET',
    url: `**/hakukohde/list**`,
    response: [],
  });

  stubHakemuspalveluLomakkeetRoute();
  stubOppijanumerorekisteriHenkiloRoute();
};
