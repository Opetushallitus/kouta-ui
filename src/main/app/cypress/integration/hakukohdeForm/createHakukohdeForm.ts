import {
  getRadio,
  selectOption,
  getCheckbox,
  fillDateTimeInput,
  fillValintakokeetSection,
  fillAsyncSelect,
  typeToEditor,
  jatka,
  getByTestId,
  paste,
  fillKieliversiotSection,
} from '#/cypress/utils';

import { prepareTest } from '#/cypress/hakukohdeFormUtils';

const lisaa = () => {
  getByTestId('lisaaButton').click({ force: true });
};

const tallenna = () => {
  getByTestId('tallennaHakukohdeButton').click();
};

const fillPohjakoulutusvaatimusSection = () => {
  getByTestId('pohjakoulutusSection').within(() => {
    getByTestId('pohjakoulutusvaatimusSelect').within(() => {
      selectOption('pohjakoulutusvaatimustoinenaste_0');
    });

    typeToEditor('Tarkenne');

    jatka();
  });
};

const fillHakuajatSection = () => {
  getByTestId('hakuajatSection').within(() => {
    getCheckbox(null).click({ force: true });
    lisaa();

    getByTestId('alkaa').within(() => {
      fillDateTimeInput({
        date: '02.04.2019',
        time: '10:45',
      });
    });

    getByTestId('paattyy').within(() => {
      fillDateTimeInput({
        date: '25.11.2019',
        time: '23:59',
      });
    });
  });
};

const fillPerustiedotSection = ({ isKorkeakoulu = false } = {}) => {
  getByTestId('perustiedotSection').within(() => {
    getByTestId('hakukohteenNimi')
      .find('input')
      .clear({ force: true })
      .pipe(paste('Hakukohteen nimi'));

    if (!isKorkeakoulu) {
      getByTestId('voiSuorittaaKaksoistutkinnon').within(() => {
        getCheckbox(null).click({ force: true });
      });
    }

    fillHakuajatSection();
    fillAlkamiskausiSection();
    fillLomakeSection();

    jatka();
  });
};

const fillLomakeSection = () => {
  getByTestId('lomakeSection').within(() => {
    getByTestId('eriHakulomake').within(() => {
      getCheckbox(null).click({ force: true });
    });

    getRadio('ataru').click({ force: true });
    selectOption('Lomake 1');
  });
};

const fillAlkamiskausiSection = () => {
  getByTestId('alkamiskausiSection').within(() => {
    getByTestId('eriAlkamiskausi').within(() => {
      getCheckbox(null).click({ force: true });
    });

    getRadio('kausi_0#1').click({ force: true });
    selectOption(new Date().getFullYear().toString());
  });
};

const fillAloituspaikatSection = ({ isKorkeakoulu = false } = {}) => {
  getByTestId('aloituspaikatSection').within(() => {
    getByTestId('aloituspaikkamaara').pipe(paste('10'));

    if (isKorkeakoulu) {
      getByTestId('ensikertalaismaara').pipe(paste('5'));
    }

    jatka();
  });
};

const fillValintaperusteenKuvausSection = () => {
  getByTestId('valintaperusteenKuvausSection').within(() => {
    selectOption('Valintaperusteen nimi');
    jatka();
  });
};

const fillLiitteetSection = () => {
  getByTestId('liitteetSection').within(() => {
    lisaa();

    getByTestId('liitelista').within(() => {
      getByTestId('tyyppi').within(() => {
        selectOption('liitetyypitamm_0');
      });

      getByTestId('nimi').find('input').pipe(paste('Nimi'));

      getByTestId('kuvaus').find('textarea').pipe(paste('Kuvaus'));

      fillDateTimeInput({
        date: '25.11.2019',
        time: '23:59',
      });

      getByTestId('toimitustapa').within(() => {
        getRadio('osoite').click({ force: true });
      });

      getByTestId('osoite').find('input').pipe(paste('Osoite'));

      getByTestId('postinumero').within(() => {
        fillAsyncSelect('0', 'Posti_0');
      });

      getByTestId('sahkoposti')
        .find('input')
        .pipe(paste('sahkoposti@email.com'));
    });

    jatka();
  });
};

const fillTilaSection = (tila = 'julkaistu') => {
  getByTestId('tilaSection').within(() => {
    getRadio(tila).check({ force: true });
  });
};

describe('createHakukohdeForm', () => {
  const organisaatioOid = '1.1.1.1.1.1';
  const hakuOid = '4.1.1.1.1.1';
  const hakukohdeOid = '1.2.3.4.5.6';
  it('should be able to create ammatillinen hakukohde', () => {
    prepareTest({
      tyyppi: 'amm',
      hakuOid,
      hakukohdeOid,
      organisaatioOid,
    });

    fillKieliversiotSection({ jatka: true });
    fillPohjakoulutusvaatimusSection();
    fillPerustiedotSection();
    fillAloituspaikatSection();
    fillValintaperusteenKuvausSection();
    fillValintakokeetSection();
    fillLiitteetSection();
    fillTilaSection();
    tallenna();

    cy.wait('@createHakukohdeRequest').then(({ request }) => {
      cy.wrap(request.body).toMatchSnapshot();
    });

    cy.location('pathname').should(
      'eq',
      `/kouta/organisaatio/${organisaatioOid}/hakukohde/${hakukohdeOid}/muokkaus`
    );
  });

  it('should be able to create korkeakoulu hakukohde', () => {
    prepareTest({
      tyyppi: 'yo',
      hakuOid,
      hakukohdeOid,
      organisaatioOid,
    });

    fillKieliversiotSection({ jatka: true });
    fillPohjakoulutusvaatimusSection();
    fillPerustiedotSection({ isKorkeakoulu: true });
    fillAloituspaikatSection({ isKorkeakoulu: true });
    fillValintaperusteenKuvausSection();
    fillValintakokeetSection();
    fillLiitteetSection();
    fillTilaSection();

    tallenna();

    cy.wait('@createHakukohdeRequest').then(({ request }) => {
      cy.wrap(request.body).toMatchSnapshot();
    });

    cy.location('pathname').should(
      'eq',
      `/kouta/organisaatio/${organisaatioOid}/hakukohde/${hakukohdeOid}/muokkaus`
    );
  });
});
