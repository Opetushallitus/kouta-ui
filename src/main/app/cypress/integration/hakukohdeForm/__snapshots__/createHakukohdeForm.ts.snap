exports[`createHakukohdeForm > should be able to create ammatillinen hakukohde #0`] =
{
  "alkamiskausiKoodiUri": "kausi_0#1",
  "alkamisvuosi": 2020,
  "aloituspaikat": 10,
  "ensikertalaisenAloituspaikat": null,
  "hakuOid": "4.1.1.1.1.1",
  "hakuajat": [
    {
      "alkaa": "2019-04-02T10:45",
      "paattyy": "2019-11-25T23:59"
    }
  ],
  "hakulomakeAtaruId": "lomake_1",
  "hakulomakeKuvaus": {},
  "hakulomakeLinkki": {},
  "hakulomaketyyppi": "ataru",
  "kaytetaanHaunAikataulua": false,
  "kaytetaanHaunAlkamiskautta": false,
  "kaytetaanHaunHakulomaketta": false,
  "kielivalinta": [
    "fi"
  ],
  "liitteet": [
    {
      "kuvaus": {
        "fi": "Kuvaus"
      },
      "nimi": {
        "fi": "Nimi"
      },
      "toimitusaika": "2019-11-25T23:59",
      "toimitusosoite": {
        "osoite": {
          "osoite": {
            "fi": "Osoite"
          },
          "postinumeroKoodiUri": "posti_0#2"
        },
        "sahkoposti": "sahkoposti@email.com"
      },
      "toimitustapa": "osoite",
      "tyyppiKoodiUri": "liitetyypitamm_0#1"
    }
  ],
  "liitteetOnkoSamaToimitusaika": false,
  "liitteetOnkoSamaToimitusosoite": false,
  "liitteidenToimitusaika": null,
  "liitteidenToimitusosoite": null,
  "liitteidenToimitustapa": null,
  "metadata": {
    "valintakokeidenYleiskuvaus": {
      "fi": "<p>Valintakokeiden kuvaus</p>"
    }
  },
  "muokkaaja": "1.2.246.562.24.62301161440",
  "nimi": {
    "fi": "Hakukohteen nimi"
  },
  "organisaatioOid": "1.1.1.1.1.1",
  "pohjakoulutusvaatimusKoodiUrit": [
    "pohjakoulutusvaatimustoinenaste_0#1"
  ],
  "pohjakoulutusvaatimusTarkenne": {
    "fi": "<p>Tarkenne</p>"
  },
  "tila": "julkaistu",
  "toinenAsteOnkoKaksoistutkinto": true,
  "toteutusOid": "2.1.1.1.1.1",
  "valintakokeet": [
    {
      "metadata": {
        "erityisjarjestelytMahdollisia": true,
        "liittyyEnnakkovalmistautumista": true,
        "ohjeetEnnakkovalmistautumiseen": {
          "fi": "<p>ohjeet ennakkovalmistautumiseen</p>"
        },
        "ohjeetErityisjarjestelyihin": {
          "fi": "<p>ohjeet erityisjärjestelyihin</p>"
        },
        "tietoja": {
          "fi": "<p>Tietoa hakijalle</p>"
        }
      },
      "nimi": {
        "fi": "nimi"
      },
      "tilaisuudet": [
        {
          "aika": {
            "alkaa": "2019-04-02T10:45",
            "paattyy": "2019-04-02T19:00"
          },
          "jarjestamispaikka": {
            "fi": "paikka"
          },
          "lisatietoja": {
            "fi": "lisatietoja"
          },
          "osoite": {
            "osoite": {
              "fi": "osoite"
            },
            "postinumeroKoodiUri": "posti_0#2"
          }
        }
      ],
      "tyyppiKoodiUri": "valintakokeentyyppi_1#1"
    }
  ],
  "valintaperusteId": "649adb37-cd4d-4846-91a9-84b58b90f928"
};

exports[`createHakukohdeForm > should be able to create korkeakoulu hakukohde #0`] =
{
  "alkamiskausiKoodiUri": "kausi_0#1",
  "alkamisvuosi": 2020,
  "aloituspaikat": 10,
  "ensikertalaisenAloituspaikat": 5,
  "hakuOid": "4.1.1.1.1.1",
  "hakuajat": [
    {
      "alkaa": "2019-04-02T10:45",
      "paattyy": "2019-11-25T23:59"
    }
  ],
  "hakulomakeAtaruId": "lomake_1",
  "hakulomakeKuvaus": {},
  "hakulomakeLinkki": {},
  "hakulomaketyyppi": "ataru",
  "kaytetaanHaunAikataulua": false,
  "kaytetaanHaunAlkamiskautta": false,
  "kaytetaanHaunHakulomaketta": false,
  "kielivalinta": [
    "fi"
  ],
  "liitteet": [
    {
      "kuvaus": {
        "fi": "Kuvaus"
      },
      "nimi": {
        "fi": "Nimi"
      },
      "toimitusaika": "2019-11-25T23:59",
      "toimitusosoite": {
        "osoite": {
          "osoite": {
            "fi": "Osoite"
          },
          "postinumeroKoodiUri": "posti_0#2"
        },
        "sahkoposti": "sahkoposti@email.com"
      },
      "toimitustapa": "osoite",
      "tyyppiKoodiUri": "liitetyypitamm_0#1"
    }
  ],
  "liitteetOnkoSamaToimitusaika": false,
  "liitteetOnkoSamaToimitusosoite": false,
  "liitteidenToimitusaika": null,
  "liitteidenToimitusosoite": null,
  "liitteidenToimitustapa": null,
  "metadata": {
    "valintakokeidenYleiskuvaus": {
      "fi": "<p>Valintakokeiden kuvaus</p>"
    }
  },
  "muokkaaja": "1.2.246.562.24.62301161440",
  "nimi": {
    "fi": "Hakukohteen nimi"
  },
  "organisaatioOid": "1.1.1.1.1.1",
  "pohjakoulutusvaatimusKoodiUrit": [
    "pohjakoulutusvaatimustoinenaste_0#1"
  ],
  "pohjakoulutusvaatimusTarkenne": {
    "fi": "<p>Tarkenne</p>"
  },
  "tila": "julkaistu",
  "toinenAsteOnkoKaksoistutkinto": false,
  "toteutusOid": "2.1.1.1.1.1",
  "valintakokeet": [
    {
      "metadata": {
        "erityisjarjestelytMahdollisia": true,
        "liittyyEnnakkovalmistautumista": true,
        "ohjeetEnnakkovalmistautumiseen": {
          "fi": "<p>ohjeet ennakkovalmistautumiseen</p>"
        },
        "ohjeetErityisjarjestelyihin": {
          "fi": "<p>ohjeet erityisjärjestelyihin</p>"
        },
        "tietoja": {
          "fi": "<p>Tietoa hakijalle</p>"
        }
      },
      "nimi": {
        "fi": "nimi"
      },
      "tilaisuudet": [
        {
          "aika": {
            "alkaa": "2019-04-02T10:45",
            "paattyy": "2019-04-02T19:00"
          },
          "jarjestamispaikka": {
            "fi": "paikka"
          },
          "lisatietoja": {
            "fi": "lisatietoja"
          },
          "osoite": {
            "osoite": {
              "fi": "osoite"
            },
            "postinumeroKoodiUri": "posti_0#2"
          }
        }
      ],
      "tyyppiKoodiUri": "valintakokeentyyppi_1#1"
    }
  ],
  "valintaperusteId": "649adb37-cd4d-4846-91a9-84b58b90f928"
};
